docs/Deprecating-Disabling-and-Removing-Formulae.md

```diff
@@ -0,0 +1,108 @@
+# Deprecating, Disabling, and Removing Formulae
+
+There are many reasons why formulae may be deprecated, disabled, or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+This general rule of thumb can be followed:
+
+- `deprecate!` should be used for formulae that should no longer be used
+- `disable!` should be used for formulae that cannot be used
+- Removal should be done with formulae that have been disabled for a long time or have a more serious issue
+
+## Deprecation
+
+If a user attempts to install a disabled formula, they will be shown a warning message but the install will succeed.
+
+A formula should be deprecated to indicate to users that the formula should not be used and may be disabled in the future. Deprecated formulae should still be able to build from source and all bottles should continue to work. Users who choose to install deprecated formulae should not have any issues.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained, or archived.
+
+To deprecate a formula, add a `deprecate!` call. This call should include a deprecation date (in the ISO 8601 format) and a deprecation reason:
+
+```ruby
+deprecate! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the project became (or will become) deprecated. If there is no clear date but the formula needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the formula will not become deprecated until that date. This can be useful if the upstream developers have indicated a date where the project will stop being supported.
+
+The `because` parameter can be set to a preset reason (using a symbol) or a custom reason. See the [Deprecate and Disable Reasons](#deprecate-and-disable-reasons) section below for more details about the `because` parameter.
+
+## Disabling
+
+If a user attempts to install a disabled formula, they will be shown an error message and the install will fail.
+
+A formula should be disabled to indicate to users that the formula cannot be used and may be removed in the future. Disabled formulae may not be able to build from source and may not have working bottles. Users who choose to attempt to install disabled formulae will likely run into issues.
+
+The most common reasons for disabling are when the formula cannot be built from source (meaning no bottles can be built), has been deprecated for a long time, the upstream repository has been removed, or the project has license issues that are not compatible with homebrew/core.
```

Comment:
In my experience we just outright remove formulae with license issues. Should we have disabled them first?

Reply:
Ah, good point. I think we `disable!` if the formula has no license and remove if it's a non-oss license. I'll clarify.